### PR TITLE
Add missing tests for API realm_*

### LINKED ISSUE #6684: Test server auth cmd: `realm_create`

### LINKED ISSUE #6685: Test server auth cmd: `realm_rename`

### LINKED ISSUE #6686: Test server auth cmd: `realm_unshare`

### LINKED ISSUE #6687: Test server auth cmd: `realm_rotate_key`

### PR DESCRIPTION
**Pull request by mmmarcos**
This PR adds missing tests for commands:
  - `realm_create`
  - `realm_rename`
  - `realm_rotate_key`
  - `realm_unshare`

Also:
- Add some `pytest.fixture` to improve test readability
- Add `common/realm.py` file with functions to patch certificates. IMO this also improves test readability because each test only patches what is needed, instead of creating certificates every time which has a lot of redundancy
- Fix test function names in `realm_share`

Closes #6684 
Closes #6685 
Closes #6686
Closes #6687

### CODE REVIEW

File: server/tests/api_v4/authenticated/test_realm_create.py [hunk 1]
```diff
@@ -24,23 +49,120 @@ async def test_authenticated_realm_create_ok(coolorg: CoolorgRpcClients, backend
...
+    "kind",
+    (
+        "dummy_data",
+        "bad_author",
+        pytest.param(
+            "author_is_outsider",
+            marks=pytest.mark.xfail(
+                reason="TODO: It seems that outsider is allowed to create realm. Not implemented?"
+            ),
+        ),
```
Comment (mmmarcos) [inline server/tests/api_v4/authenticated/test_realm_create.py:85]: Not sure if this is OK or if the implementation is missing.

File: server/tests/api_v4/authenticated/test_realm_rename.py [hunk 2]
```diff
@@ -294,28 +272,95 @@ async def test_authenticated_realm_rename_bad_key_index(
...
+
+
+@pytest.mark.xfail(
+    reason="TODO: realm_rename never returns RepRequireGreaterTimestamp. Not implemented?"
+)
+@pytest.mark.parametrize(
+    "timestamp_offset",
+    (pytest.param(0, id="same_timestamp"), pytest.param(1, id="previous_timestamp")),
+)
+async def test_authenticated_realm_rename_require_greater_timestamp(
```
Comment (mmmarcos) [inline server/tests/api_v4/authenticated/test_realm_rename.py:306]: Not sure if there is a problem with the test or the implementation is missing.

File: server/tests/api_v4/authenticated/test_realm_create.py (hunk 1 above)
Comment (mmmarcos) [reply server/tests/api_v4/authenticated/test_realm_create.py]: This is actually expected behavior for outsider profile: realm_create is authorized, but not realm_share.

File: server/tests/api_v4/authenticated/test_realm_rename.py (hunk 2 above)
Comment (touilleMan) [reply server/tests/api_v4/authenticated/test_realm_rename.py:306]: good catch ! this is an error in the implementation

File: server/tests/api_v4/authenticated/test_realm_rename.py [hunk 3]
```diff
@@ -254,37 +251,27 @@ async def test_authenticated_realm_rename_bad_key_index(
...
-        author=coolorg.alice.device_id,
-        timestamp=t1,
-        realm_id=wksp_id,
-        key_index=bad_key_index,
-        encrypted_name=b"<encrypted name>",
+    certif = patch_realm_name_certificate(
+        alice_name_certificate, realm_id=wksp_id, key_index=bad_key_index
     )
-
+    print(certif)
```
[the remark this answers is not in the document]
Comment (mmmarcos) [reply server/tests/api_v4/authenticated/test_realm_rename.py]: removed

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**